### PR TITLE
Pontential duplicates from task page

### PR DESCRIPTION
**Pull request by iravisankar**

### CODE REVIEW

File: src/app/core/services/tasks.service.ts [hunk 1]
```diff
@@ -342,6 +356,38 @@ export class TasksService {
...
+            duplicatesSets.map((value) => value.transaction_ids)
+          );
+
+          const task = [
+            {
+              isAmountHidden: true,
+              count: duplicatesSets.length,
+              header: `${duplicateIds.length} Potential duplicates`,
+              subheader: `we detected ${duplicateIds.length} expenses which may be duplicates`,
+              icon: TaskIcon.REPORT,
```
Comment (Dimple16) [inline src/app/core/services/tasks.service.ts:374]: are we using the same icon for duplicates?

File: src/app/fyle/potential-duplicates/potential-duplicates.page.scss [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  &--body {
+    background: $white;
+    height: 100%;
+  }
+
+  &--navigation {
+    display: flex;
+    height: 60px;
+    justify-content: space-between;
+    padding-top: 20px;
```
Comment (Dimple16) [inline src/app/fyle/potential-duplicates/potential-duplicates.page.scss:13]: `padding: 20px 0;`

File: src/app/fyle/potential-duplicates/potential-duplicates.page.scss [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    padding-bottom: 20px;
+
+    &-button {
+      color: $brand-primary;
+      font-size: 16px;
+      padding: 10px;
+      width: 20%;
+      text-align: center;
```
Comment (Dimple16) [inline src/app/fyle/potential-duplicates/potential-duplicates.page.scss:21]: use global class for `text-center`

File: src/app/fyle/potential-duplicates/potential-duplicates.page.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+export class PotentialDuplicatesPage implements OnInit {
+  duplicatesSet$: Observable<any>;
+
+  loadData$: BehaviorSubject<any>;
+
+  selectedSet = 0;
+
+  duplicatesSetCount: number;
+
+  duplicatesSetData;
```
Comment (Dimple16) [inline src/app/fyle/potential-duplicates/potential-duplicates.page.ts:20]: is this of a specific type?

File: src/app/shared/components/expenses-card/expenses-card.component.scss [hunk 5]
```diff
@@ -297,4 +297,10 @@
     text-align: center;
     padding: 4px;
   }
+
+  &--dismiss {
+    margin-top: -10px;
```
Comment (Dimple16) [inline src/app/shared/components/expenses-card/expenses-card.component.scss:302]: `margin: -10px 0 0 12px;`

**Review by Dimple16 — CHANGES_REQUESTED** [review]

File: src/app/core/services/tasks.service.ts (hunk 1 above)
Comment (iravisankar) [reply src/app/core/services/tasks.service.ts:374]: same icon used in design

File: src/app/fyle/potential-duplicates/potential-duplicates.page.scss [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    background: $white;
+    height: 100%;
+  }
+
+  &--navigation {
+    display: flex;
+    height: 60px;
+    justify-content: space-between;
+    padding: 20px 0;
+    padding-bottom: 20px;
```
Comment (Dimple16) [inline src/app/fyle/potential-duplicates/potential-duplicates.page.scss:14]: remove either of the padding

File: src/app/fyle/potential-duplicates/potential-duplicates.page.scss [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  &--container {
+    height: 88%;
+    width: 93%;
+    margin: 10px auto 20px auto;
+    background: $off-white;
+    border: 1px solid $grey;
+    box-sizing: border-box;
+    border-radius: 15px;
```
Comment (Dimple16) [inline src/app/fyle/potential-duplicates/potential-duplicates.page.scss:39]: does 14 or 16px work?

File: src/app/fyle/potential-duplicates/potential-duplicates.page.scss [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+    align-content: center;
+    border-radius: 20px;
+  }
+
+  &--header {
+    font-size: 18px;
+    line-height: 23px;
+    color: $pure-black;
+    text-align: center;
```
Comment (Dimple16) [inline src/app/fyle/potential-duplicates/potential-duplicates.page.scss:48]: use `text-center`

Comment (Dimple16) [reply src/app/fyle/potential-duplicates/potential-duplicates.page.scss:48]: global class

File: src/app/fyle/potential-duplicates/potential-duplicates.page.ts [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+      .subscribe((duplicatesSetCount) => {
+        this.duplicatesSetCount = duplicatesSetCount;
+      });
+
+    this.loadData$ = new BehaviorSubject({});
+
+    this.duplicatesSet$ = this.loadData$.pipe(
+      switchMap(() =>
+        this.handleDuplicates.getDuplicatesSet().pipe(
+          tap((duplicatesSets) => {
```
[the remark this answers is not in the document]
Comment (iravisankar) [reply src/app/fyle/potential-duplicates/potential-duplicates.page.ts:40]: Storing in duplicatesSetData on every trigger to use in other places.

File: src/app/core/models/v2/duplicate-sets.model.ts [hunk 10]
```diff
@@ -0,0 +1,4 @@
+export interface DuplicateSets {
+  fields: any[];
```
Comment (Dimple16) [inline src/app/core/models/v2/duplicate-sets.model.ts:2]: any?

File: www/.gitignore [hunk 11]
```diff
@@ -1,4 +0,0 @@
-# Ignore everything in this directory
```
Comment (Dimple16) [inline www/.gitignore]: any reason this is removed?

**Review by Dimple16 — CHANGES_REQUESTED** [review]

File: www/.gitignore (hunk 11 above)
Comment (iravisankar) [reply www/.gitignore]: got an error on Appflow to add `www` folder. so added it, but found the real issue...,  so removed it now...

File: src/app/core/services/handle-duplicates.service.ts [hunk 12]
```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { ApiService } from './api.service';
+@Injectable({
+  providedIn: 'root',
+})
+export class HandleDuplicatesService {
+  constructor(private apiService: ApiService) {}
+
+  getDuplicatesSet() {
```
Comment (Dimple16) [inline src/app/core/services/handle-duplicates.service.ts:10]: should we add a return type?

**Review by Dimple16 — CHANGES_REQUESTED** [review]

File: src/app/core/services/handle-duplicates.service.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    return this.apiService.get('/transactions/duplicates/sets');
+  }
+
+  dismissAll(duplicateSetTransactionIds: string[], transactionIds: string[]) {
```
[the remark this answers is not in the document]
Comment (iravisankar) [reply src/app/core/services/handle-duplicates.service.ts:14]: it does not return any value

**Review by Dimple16 — APPROVED** [review]

**Pull request closed without merging**